cranelift/codegen/src/machinst/buffer.rs

```diff
@@ -998,7 +1027,13 @@ impl<I: VCodeInst> MachBuffer<I> {
         data: &[u8],
         max_distance: CodeOffset,
     ) {
-        let deadline = self.cur_offset() + max_distance;
+        trace!(
+            "defer_constant: eventually emit {} bytes aligned to {} at label {:?}",
+            data.len(),
+            align,
+            label
+        );
+        let deadline = self.cur_offset().saturating_add(max_distance);
         self.island_worst_case_size += data.len() as CodeOffset;
         self.island_worst_case_size &= !(I::LabelUse::ALIGN - 1);
```

Comment:
Pre-existing, but I think this upward-padding alignment might be written incorrectly; it should be `(x + align - 1) & ~(align - 1)`, not `x & ~(align - 1)`. Eek. (On x64 `LabelUse::ALIGN` is `1` so no big deal, but...)  Could you change this to:

```
self.island_worst_case_size = (self.island_worst_case_size + I::LabelUse::ALIGN - 1) & !(I::LabelUse::ALIGN - 1);
```

or better, use an alignment helper function that we must surely already have somewhere?

Reply:
I didn't find the helper, but I was thinking that if we made `CodeOffset` an actual struct we could add an `align()` there. Also, are we sure we don't need to do saturating addition in a few more places in this code?

Reply:
That's actually a very nice idea!

Re: saturating addition -- perhaps, yes; we should audit this code. I'll take a look.